From 0d7b90880de6653bf6ee033420dffa63fbaf6c9c Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Mon, 26 Jun 2017 21:08:08 +0800
Subject: [PATCH 1048/1345] arm64: dts: marvell: Update Armada 3700 IO clocks

commit  7f85e4dcf6fc745b23512313a0f9b96f68d5e127 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Because of the new clcok driver, the old clcok node
  needs to be updated to make IO driver can get the clock.
- This patch just update the clock node in IO node to make
  to IO driver work well.

Change-Id: If0ff136493ef3e7f50ab5b4df0f8ada97aaa05a3
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40826
Reviewed-by: Victor Gu <xigu@marvell.com>
Tested-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi |   29 +++++++-------------------
 1 file changed, 8 insertions(+), 21 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index c7d4874..e848d16 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -96,14 +96,6 @@
 		};
 	};
 
-	clocks {
-		osc: osc {
-			compatible = "fixed-clock";
-			#clock-cells = <0>;
-			clock-frequency = <25000000>;
-		};
-	};
-
 	psci {
 		compatible = "arm,psci-0.2";
 		method = "smc";
@@ -174,6 +166,7 @@
 			uart0: serial@12000 {
 				compatible = "marvell,armada-3700-uart";
 				reg = <0x12000 0x200>;
+				clocks = <&xtalclk>;
 				interrupts = <GIC_SPI 11 IRQ_TYPE_LEVEL_HIGH>;
 				interrupt-names = "irq_sum";
 				status = "disabled";
@@ -331,7 +324,7 @@
 				compatible = "marvell,armada-3700-wdt";
 				reg = <0xd064 0x4>,
 				      <0x8300 0x40>;
-				clocks = <&osc>;
+				clocks = <&xtalclk>;
 			};
 
 			npinctrl: north-bridge-pinctrl@13830 {
@@ -364,13 +357,7 @@
 				compatible = "marvell,armada-37xx-cpu-pm-clk";
 				reg = <0x14018 0x1C>;
 				#clock-cells = <1>;
-				clocks = <&core_clk 4>;
-			};
-
-			sgateclk: south-bridge-gate-clk@18014 {
-				compatible = "marvell,armada-3700-south-bridge-gating-clock";
-				reg = <0x18014 0x4>;
-				#clock-cells = <1>;
+				clocks = <&nb_periph_clk 16>;
 			};
 
 			spinctrl: south-bridge-pinctrl@18830 {
@@ -389,7 +376,7 @@
 				/* 0: 0x50000: USB 3.0 Device port 0: DEV_INFO_REG(0:15 - version_id) */
 				reg = <0x50000 0x2000>;
 				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&sgateclk 17>;
+				clocks = <&sb_periph_clk 12>;
 				status = "disabled";
 			};
 
@@ -397,7 +384,7 @@
 				compatible = "marvell,mv-udc";
 				reg = <0x54100 0x2000>;
 				interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&sgateclk 17>;
+				clocks = <&sb_periph_clk 12>;
 				status = "disabled";
 			};
 			usb3: usb@58000 {
@@ -487,7 +474,7 @@
 				compatible = "marvell,armada-3700-ahci";
 				reg = <0xe0000 0x178>;
 				interrupts = <GIC_SPI 27 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&ngateclk 3>;
+				clocks = <&nb_periph_clk 1>;
 				status = "disabled";
 			};
 
@@ -513,7 +500,7 @@
 			bm: bm@c0000 {
 				compatible = "marvell,armada-380-neta-bm";
 				reg = <0xc0000 0xac>;
-				clocks = <&sgateclk 9>;
+				clocks = <&sb_periph_clk 9>;
 				internal-mem = <&bm_bppi>;
 				status = "disabled";
 			};
@@ -521,7 +508,7 @@
 			bm_bppi: bm-bppi {
 				compatible = "mmio-sram";
 				reg = <0xc8000 0x1000>;
-				clocks = <&sgateclk 9>;
+				clocks = <&sb_periph_clk 9>;
 				no-memory-wc;
 				status = "disabled";
 			};
-- 
1.7.9.5

